aten/src/ATen/native/Memory.cpp

```diff
@@ -12,7 +12,7 @@ Tensor pin_memory(const Tensor& self) {
     AT_ERROR("cannot pin '", self.type().toString(), "' only CPU memory can be pinned");
   }
   auto* allocator = detail::getCUDAHooks().getPinnedMemoryAllocator();
-  auto tensor = self.type().tensorWithAllocator(self.sizes(), self.strides(), allocator);
+  auto tensor = self.dispatch_type().tensorWithAllocator(self.sizes(), self.strides(), allocator);
```

Comment:
Another one where it would be nice to stop using the factory function on type.

Reply:
Can't be done yet, but in the future :)